[charts] Move the line item highligh into a dedicated component (#10117)
mui/mui-x: docs/data/charts/lines/lines.md

{{"demo": "InterpolationDemo.js", "hideToolbar": true, "bg": "inline"}}

### Optimization

To show mark elements, use `showMark` series property.
It accepts a boolean or a callback.
The next example shows how to use it to display only one mark every two data points.

> When the value is highlighted, an additional mark is rendered over the other ones.

Reviewer: Isn't the current sentence a bit misleading?
Based on how it works, it seems that a mark is rendered only for a given value, which is being highlighted.
```suggestion
When a value is highlighted, a mark is rendered for that given value.
```

Author: Yes, but if you set `showMark=true` for N data points, you get N marks + the highlighted one

Reviewer: Hmmm... Gotcha. 🤔 
Then maybe the explanation could be even more clear. 🤷

Author: What about 

```suggestion
When a value is highlighted, a mark is rendered for that given value.
If the charts already have some marks (due to `showMark=true`) the highlight one will be on top of others.
```

Reviewer: IMHO, the suggestion brings more clarity. 👍 💯

> This behavior can be removed with series property `disableHighlight` or at the scale of the line chart with prop `disableLineItemHighlight`.

Reviewer: ```suggestion
This behavior can be removed with the `disableHighlight` series property or at the root of the line chart with a `disableLineItemHighlight` prop.
```
